src/services/services.ts

```diff
@@ -584,7 +584,17 @@ namespace ts {
             forEachUnique(declarations, declaration => {
                 const inheritedDocs = findBaseOfDeclaration(checker, declaration, symbol => symbol.getDocumentationComment(checker));
                 // TODO: GH#16312 Return a ReadonlyArray, avoid copying inheritedDocs
-                if (inheritedDocs) doc = doc.length === 0 ? inheritedDocs.slice() : inheritedDocs.concat(lineBreakPart(), doc);

```

Comment:
I don't think you need to write your own loop; `findBaseOfDeclaration` allows you to skip base declarations by returning undefined, so you can maintain `seen: Set<Symbol>` to allow you to skip a symbol that's been seen before.

Reply:
Also, the outer forEachUnique might be redundant with this fix in place. Might be worth running tests once with it removed to see.

Reply:
GENUIS!